### DISCUSSION TITLE
Override used compiler for only one package

### DISCUSSION THREAD

**Discussion by gracicot** [microsoft/vcpkg#24679] [Q&A]
Is there a way to override which compiler one package uses? I'm trying to build a project that uses vcpkg and depends on boost, but `boost-build` don't work on `arm64-osx` if you use GCC, it only works with clang. My project only work with GCC and GCC compiled libraries, but `boost-build` is a tool, not a library. Since I'm not linking against boost build and I simply use it so I can install `boost-system`, I would like to build `boost-build` using Clang, and any other library can be built using GCC.

Is there a way I can do this? The thing I'm doing right now is to first setup an environment using Clang, then do the following:

 - `vcpkg install boost-build`,
 - Copy the directory `vcpkg/installed/arm64-osx/tools/boost-build` into a temporary directory (let's say `~/b2-install/boost-build`)
 - Deleting the vcpkg cache and buildtrees and installed packages
 - Setup environment with GCC
 - First run CMake with my project and vcpkg toolchain, and let it fail
 - Delete the `./build/vcplg_installed/arm64-osx/tools/boost-build`
 - Copy `~/b2-install/boost-build` into `./build/vcplg_installed/arm64-osx/tools/`
 - Run CMake again, and it will succeed

If there would be a way to simply override the compiler used for `boost-build`, it would be perfect.

**Comment by Neumann-A** [comment]
You can define the used toolchain per port. This however requires a custom triplet. I use it in:
https://github.com/Neumann-A/my-vcpkg-triplets/blob/master/x64-windows-llvm.cmake
https://github.com/Neumann-A/my-vcpkg-triplets/blob/master/x64-windows-llvm.port.cmake
and switch between `clang-cl` and `cl` in some cases. (Windows is more complex than linux, you can ignore most of the stuff in the triplet)

**Reply by gracicot** [reply]
Thank you very much! Let me try this and I'll get back to you
